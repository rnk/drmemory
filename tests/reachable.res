# **********************************************************
# Copyright (c) 2011 Google, Inc.  All rights reserved.
# Copyright (c) 2009-2010 VMware, Inc.  All rights reserved.
# **********************************************************
#
# Dr. Memory: the memory debugger
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; 
# version 2.1 of the License, and no later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Library General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#
Error #1: UNINITIALIZED READ
cs2bug.cpp:83
Error #2: UNADDRESSABLE ACCESS: writing 4 byte(s)
cs2bug.cpp:91
%if !CYGWIN
Error #3: INVALID HEAP ARGUMENT: wrong free/delete/delete[] called
cs2bug.cpp:93
memory was allocated here:
cs2bug.cpp:87
Error #4: INVALID HEAP ARGUMENT: wrong free/delete/delete[] called
cs2bug.cpp:176
memory was allocated here:
cs2bug.cpp:174
Error #5: INVALID HEAP ARGUMENT: wrong free/delete/delete[] called
cs2bug.cpp:179
memory was allocated here:
cs2bug.cpp:177
Error #6: INVALID HEAP ARGUMENT: wrong free/delete/delete[] called
cs2bug.cpp:182
memory was allocated here:
cs2bug.cpp:180
Error #7: INVALID HEAP ARGUMENT: wrong free/delete/delete[] called
cs2bug.cpp:185
memory was allocated here:
cs2bug.cpp:183
%endif
%if CYGWIN
# i#589: no syms so can't find some errors due to asymmetric inlines (xref i#123)
Error #3: INVALID HEAP ARGUMENT: wrong free/delete/delete[] called
cs2bug.cpp:179
memory was allocated here:
cs2bug.cpp:177
Error #4: INVALID HEAP ARGUMENT: wrong free/delete/delete[] called
cs2bug.cpp:182
memory was allocated here:
cs2bug.cpp:180
Error #5: INVALID HEAP ARGUMENT: wrong free/delete/delete[] called
cs2bug.cpp:185
memory was allocated here:
cs2bug.cpp:183
Error #6: LEAK 4 direct bytes + 0 indirect bytes
cs2bug.cpp:82
%endif
%if UNIX
Error #8: LEAK 4 direct bytes + 19 indirect bytes
cs2bug.cpp:155
Error #9: LEAK 4 direct bytes + 0 indirect bytes
cs2bug.cpp:82
%endif
%if WINDOWS
# FIXME PR 587093: string code disabled for now
Error #8: LEAK 4 direct bytes + 0 indirect bytes
cs2bug.cpp:82
%endif
# ensure reachable leaks are printed, and after regular leaks
REACHABLE LEAK
REACHABLE LEAK
REACHABLE LEAK
