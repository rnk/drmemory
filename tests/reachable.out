# **********************************************************
# Copyright (c) 2011 Google, Inc.  All rights reserved.
# Copyright (c) 2009-2010 VMware, Inc.  All rights reserved.
# **********************************************************
#
# Dr. Memory: the memory debugger
#
# This library is free software; you can redistribute it and/or
# modify it under the terms of the GNU Lesser General Public
# License as published by the Free Software Foundation; 
# version 2.1 of the License, and no later version.
#
# This library is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
# Library General Public License for more details.
#
# You should have received a copy of the GNU Lesser General Public
# License along with this library; if not, write to the Free Software
# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
#
%if WINDOWS
# if this starts not matching on other platforms/compiler just remove the print!
size=4 capacity=15
%endif
%if UNIX
size=4 capacity=4
%endif
getval: childAB
myfunc: parB
hi
bye
~~Dr.M~~ ERRORS FOUND:
~~Dr.M~~       1 unique,     1 total unaddressable access(es)
~~Dr.M~~       1 unique,     1 total uninitialized access(es)
%if CYGWIN
# i#589: no syms so can't find due to asymmetric inlines (xref i#123)
~~Dr.M~~       3 unique,     3 total invalid heap argument(s)
%endif
%if !CYGWIN
~~Dr.M~~       5 unique,     5 total invalid heap argument(s)
%endif
~~Dr.M~~       0 unique,     0 total warning(s)
%if UNIX
~~Dr.M~~       2 unique,     2 total,     27 byte(s) of leak(s)
%endif
%if WINDOWS
# FIXME PR 587093: string code disabled for now
~~Dr.M~~       1 unique,     1 total,      4 byte(s) of leak(s)
%endif
~~Dr.M~~       0 unique,     0 total,      0 byte(s) of possible leak(s)
still-reachable allocation(s)
